packages/amplify-frontend-android/lib/amplify-config-helper.js

```diff
@@ -187,6 +188,62 @@ function constructStorage(metadata, amplifyConfig) {
   }
 }
 
+function constructGeo(metadata, amplifyConfig) {
+  const categoryName = 'geo';
+  const pluginName = 'awsLocationGeoPlugin';
+  let geoRegion = metadata.providers.awscloudformation.Region;
+  if (metadata[categoryName] && Object.keys(metadata[categoryName]).length > 0) {
```

Comment:
nit: I prefer to early return rather than nest everything within an if statement (applies to the logic in the other frontend config generators as well)